frontend/src/components/widgets/DataFrame/columns/NumberColumn.test.ts

```diff
@@ -0,0 +1,201 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { GridCellKind, NumberCell } from "@glideapps/glide-data-grid"
+
+import { DataType, Type as ArrowType } from "src/lib/Quiver"
+
+import { BaseColumnProps, isErrorCell } from "./utils"
+import NumberColumn, { NumberColumnParams } from "./NumberColumn"
+
+const MOCK_FLOAT_ARROW_TYPE: ArrowType = {
+  pandas_type: "float64",
+  numpy_type: "float64",
+}
+
+const MOCK_INT_ARROW_TYPE: ArrowType = {
+  pandas_type: "int64",
+  numpy_type: "int64",
+}
+
+const MOCK_UINT_ARROW_TYPE: ArrowType = {
+  pandas_type: "uint64",
+  numpy_type: "uint64",
+}
+
+const NUMBER_COLUMN_TEMPLATE: Partial<BaseColumnProps> = {
+  id: "1",
+  title: "Number column",
+  indexNumber: 0,
+  isEditable: false,
+  isHidden: false,
+  isIndex: false,
+  isStretched: false,
+}
+
+function getNumberColumn(
+  arrowType: ArrowType,
+  params?: NumberColumnParams
+): ReturnType<typeof NumberColumn> {
+  return NumberColumn({
+    ...NUMBER_COLUMN_TEMPLATE,
+    arrowType,
+    columnTypeMetadata: params,
+  } as BaseColumnProps)
+}
+
+describe("NumberColumn", () => {
+  it("creates a valid column instance", () => {
+    const mockColumn = getNumberColumn(MOCK_FLOAT_ARROW_TYPE)
+    expect(mockColumn.kind).toEqual("number")
+    expect(mockColumn.title).toEqual(NUMBER_COLUMN_TEMPLATE.title)
+    expect(mockColumn.id).toEqual(NUMBER_COLUMN_TEMPLATE.id)
+    expect(mockColumn.isEditable).toEqual(NUMBER_COLUMN_TEMPLATE.isEditable)
+    expect(mockColumn.sortMode).toEqual("smart")
+
+    const mockCell = mockColumn.getCell("1.234")
+    expect(mockCell.kind).toEqual(GridCellKind.Number)
+    expect((mockCell as NumberCell).displayData).toEqual("1.234")
+    expect((mockCell as NumberCell).data).toEqual(1.234)
+  })
+
+  it("alignes numbers to the right", () => {
+    const mockColumn = getNumberColumn(MOCK_FLOAT_ARROW_TYPE)
+    const mockCell = mockColumn.getCell("1.123")
+    expect(mockCell.contentAlign).toEqual("right")
+  })
+
+  it.each([
+    [true, 1],
+    [false, 0],
+    ["4.12", 4.12],
+    ["-4.12", -4.12],
+    ["4", 4],
+    [1.3122, 1.3122],
+    ["1,212.12", 1212.12],
+    [".1312314", 0.1312314],
+    [null, null],
+    [undefined, null],
+    ["", null],
+  ])(
+    "supports float64 value (%p parsed as %p)",
+    (input: DataType | null | undefined, value: number | null) => {
+      const mockColumn = getNumberColumn(MOCK_FLOAT_ARROW_TYPE)
+      const cell = mockColumn.getCell(input)
+      expect(mockColumn.getCellValue(cell)).toEqual(value)
+    }
+  )
+
+  it.each([
+    [100, 100],
+    [-100, -100],
+    ["4", 4],
+    ["4.12", 4],
```

Comment:
nit: this string test case can be removed

Reply:
Hmm, I added the `4.12` and `4.61` tests here to test that it is not doing any unexpected rounding. E.g., `4.61` could also lead to `5` depending on how the number is truncated. The `4.12` is probably not super relevant here, but I will keep that in for this reason.